packages/app/src/vis-packs/nexus/utils.ts

```diff
@@ -124,6 +124,53 @@ export function findAssociatedDatasets(
   });
 }
 
+function parseAxesList(dsetList: unknown): string[] {
+  if (typeof dsetList !== 'string') {
+    assertArray<string>(dsetList);
+    return dsetList;
+  }
+
+  if (dsetList.includes(':')) {
+    return dsetList.split(':');
+  }
+
+  if (dsetList.includes(',')) {
+    return dsetList.split(',');
+  }
+
+  return [dsetList];
+}
+
+export function findOldStyleAxesDatasets(
+  group: GroupWithChildren,
+  signal: Dataset,
+  attrValuesStore: AttrValuesStore
+): NumArrayDataset[] {
+  const axesList = attrValuesStore.getSingle(signal, 'axes') || [];
+  const axesNames = parseAxesList(axesList);
```

Comment:
I don't believe the spec mentions that `axes` can be an array. To simplify things, I would just return an empty array if the value of the attribute is not a string (with a type guard, not an assertion, so it doesn't crash). This would ensure that we always end up with an array of strings without having to double check the type of the `axes` attribute.

Reply:
Good catch. I think I was still considering treating the `axes` of version 3 but it is not relevant here.